clippy_lints/src/operators/arithmetic.rs

```diff
@@ -50,36 +52,82 @@ impl Arithmetic {
         span_lint(cx, ARITHMETIC, expr.span, "arithmetic detected");
         self.expr_span = Some(expr.span);
     }
+
+    /// Checks assign operators (+=, -=, *=, /=) already handled by the CTFE.
+    fn has_valid_assign_op(op: &Spanned<hir::BinOpKind>, rhs: &hir::Expr<'_>) -> bool {
+        match op.node {
+            hir::BinOpKind::Div | hir::BinOpKind::Rem => match rhs.kind {
+                hir::ExprKind::Lit(_) => true,
+                hir::ExprKind::Unary(hir::UnOp::Neg, expr) => {
+                    let hir::ExprKind::Lit(lit) = &expr.kind else {
+                        return true;
+                    };
+                    if let ast::LitKind::Int(1, _) = lit.node {
+                        return false;
+                    }
+                    true
+                },
+                _ => false,
+            },
+            _ => false,
+        }
+    }
+
+    /// Checks binary operators (+, -, *, /) already handled by the CTFE.
+    fn has_valid_op(lhs: &hir::Expr<'_>, lhs_refs: Ty<'_>, rhs: &hir::Expr<'_>, rhs_refs: Ty<'_>) -> bool {
+        let lhs_is_integral = lhs_refs.is_integral();
+        let lhs_is_literal = matches!(lhs.kind, hir::ExprKind::Lit(_));
+        let rhs_is_integral = rhs_refs.is_integral();
+        let rhs_is_literal = matches!(rhs.kind, hir::ExprKind::Lit(_));
+        lhs_is_integral && lhs_is_literal && rhs_is_integral && rhs_is_literal
+    }
+
+    /// Manages when the lint should be triggered.
+    fn manage_bin_ops(
+        &mut self,
+        cx: &LateContext<'_>,
+        expr: &hir::Expr<'_>,
+        op: &Spanned<hir::BinOpKind>,
+        lhs: &hir::Expr<'_>,
+        rhs: &hir::Expr<'_>,
+    ) {
+        let (
+            hir::BinOpKind::Add
+            | hir::BinOpKind::Sub
+            | hir::BinOpKind::Mul
+            | hir::BinOpKind::Div
+            | hir::BinOpKind::Rem
+            | hir::BinOpKind::Shl
+            | hir::BinOpKind::Shr
+        ) = op.node else {
+            return;
+        };
+        if self.is_allowed_ty(cx, lhs) || self.is_allowed_ty(cx, rhs) {
+            return;
+        }
+        let lhs_refs = cx.typeck_results().expr_ty(lhs).peel_refs();
+        let rhs_refs = cx.typeck_results().expr_ty(rhs).peel_refs();
+        if lhs_refs.is_floating_point() && rhs_refs.is_floating_point() {
+            self.issue_lint(cx, expr);
+            return;
+        }
```

Comment:
Should be able to remove this check if both `has_valid_assign_op` and `has_valid_op` are checking the type is integeral